src/main/java/at/hannibal2/skyhanni/features/garden/farming/LaneswitchNotification.kt

```diff
@@ -0,0 +1,105 @@
+
+class LaneswitchNotification {
+
+    private val config get() = GardenAPI.config.laneswitch
+
+    private var blocksPerSecond = MovementSpeedDisplay.currentBPS
+    private var lastBlocksPerSecond = 0.0
+    private var lastPos = LorenzVec(0, 0, 0)
+    private var lastLaneSwitch = SimpleTimeMark.farPast()
+    private var lastWarning = SimpleTimeMark.farPast()
+
+    companion object {
+        private val config get() = GardenAPI.config.laneswitch
+
+        @JvmStatic
+        fun playUserSound() {
+            SoundUtils.createSound(
+                config.notification.sound.notificationSound,
+                config.notification.sound.notificationPitch,
+            ).playSound()
+        }
+    }
+
+    private fun switchPossibleInTime(from: LorenzVec, to: LorenzVec, speed: Double, time: Int): Boolean {
+        return from.distance(to) <= speed * time
+    }
+
+    private fun getFarmBounds(plotIndex: Int, playerPos: LorenzVec, lastPos: LorenzVec): List<LorenzVec>? {
+        val xVelocity = playerPos.x - lastPos.x
+        val zVelocity = playerPos.z - lastPos.z
+        if (xVelocity.absoluteValue > zVelocity.absoluteValue) {
+            val xValueMin = if (plotIndex - 1 == -1 || !plots[plotIndex - 1].unlocked || plots.indexOf(plots[plotIndex - 1]) == 12)
+                plots[plotIndex].box.minX else plots[plotIndex - plotIndex % 5].box.minX
+            val xValueMax = if ((plotIndex + 1) % 5 == 0 || plotIndex + 1 == 25 || !plots[plotIndex + 1].unlocked || plots.indexOf(plots[plotIndex + 1]) == 12)
+                plots[plotIndex].box.maxX else plots[plotIndex + (4 - (plotIndex % 5))].box.maxX
+
+            return listOf(
+                LorenzVec(xValueMin, playerPos.y, playerPos.z), LorenzVec(xValueMax, playerPos.y, playerPos.z)
+            )
+        } else if (xVelocity.absoluteValue < zVelocity.absoluteValue) {
+            val zValueTop = if (plotIndex - 1 == -1 || plotIndex - 5 < 0 || !plots[plotIndex - 5].unlocked || plots.indexOf(plots[plotIndex - 5]) == 12)
+                plots[plotIndex].box.minZ else plots[plotIndex - 5].box.minZ
+            val zValueBottom = if ((plotIndex + 1) % 5 == 0 || plotIndex + 5 > 24 || !plots[plotIndex + 5].unlocked || plots.indexOf(plots[plotIndex + 5]) == 12)
+                plots[plotIndex].box.maxZ else plots[plotIndex + 5].box.maxZ
+
+            return listOf(
+                LorenzVec(playerPos.x, playerPos.y, zValueTop), LorenzVec(playerPos.x, playerPos.y, zValueBottom)
+            )
+        } else if (xVelocity.absoluteValue == 0.0 && zVelocity.absoluteValue == 0.0) {
+            return null
+        }
+        return null
+    }
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!config.enabled || !GardenAPI.inGarden()) return
+        if (!plots.any { it.unlocked } && lastWarning.passedSince() >= 30.seconds) {
+            sendTitle(LorenzColor.RED.getChatColor() + "Plots aren't loaded, open 'Configure Plots'", 2.seconds)
```

Comment:
the message appears ony shortly and rarely and the message doesnt include information what feature this is. maybe send a chat message instead and also include the feature name. then allow clicking on the chat message to open the settings to disable the feature, for the case whe n the user accidetnaly enables the feature and wants it gone quickly